### PR TITLE
Docker setup

### LINKED ISSUE #321: Implement Docker Setup
### Objective
Create a Docker setup for running the Django backend and React frontend.

### Background
Our current separate running of Django and React leads to environment inconsistencies and have to install the environment locally. Dockerizing both can streamline development and setup processes.

### Requirements
1. **Docker Configuration**: Create a Dockerfile and docker-compose.yml to:
   - Run Django with Daphne as the ASGI server.
   - Serve the React frontend, ensuring communication with the backend.
2. **Environment Variables**: Implement necessary variables for Django and React.
3. **Documentation**: Update README with Docker setup instructions.
4. **Testing**: Validate that both Django (including WebSocket functionality) and React work in the Docker environment.

### PR DESCRIPTION
**Pull request by blu3eee**
- Added docker setup for backend, frontend, and redis server (redis server for messaging feature using websocket and django channels)

This PR resolves #321

### CODE REVIEW

**Review by misslame — APPROVED** [review]
Looks good

**Review by misslame — APPROVED** [review]

**Comment by misslame** [comment]
yay

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Copy the current directory contents into the container at /code
+COPY Pipfile Pipfile.lock /code/
+
+# Install Python dependencies
+RUN pipenv install
+
+# Copy everything to /code folder
+COPY . /code/
+RUN chmod +x /code/entrypoint.sh
+RUN dos2unix /code/entrypoint.sh
```
Comment (joshgarde) [inline Dockerfile:26]: Might want to consider switching these line endings to Unix-style

**Review by joshgarde — APPROVED** [review]
Just one nitpick, but otherwise good to go

**Pull request merged**